cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/sampling/KafkaWriteOnlySampleStore.java

```diff
@@ -0,0 +1,275 @@
+
+      // New topics
+      if (_partitionMetricSampleStoreTopic != null) {
+        NewTopic partitionSampleStoreNewTopic =
+            wrapTopic(_partitionMetricSampleStoreTopic, _partitionSampleStoreTopicPartitionCount, replicationFactor,
+                partitionSampleRetentionMs);
+        ensureTopicCreated(adminClient, partitionSampleStoreNewTopic);
+      }
+
+      if (_brokerMetricSampleStoreTopic != null) {
+        NewTopic brokerSampleStoreNewTopic =
+            wrapTopic(_brokerMetricSampleStoreTopic, _brokerSampleStoreTopicPartitionCount, replicationFactor,
+                brokerSampleRetentionMs);
+        ensureTopicCreated(adminClient, brokerSampleStoreNewTopic);
+      }
+    } finally {
+      KafkaCruiseControlUtils.closeAdminClientWithTimeout(adminClient);
+    }
+  }
+
+  protected void ensureTopicCreated(AdminClient adminClient, NewTopic sampleStoreTopic) {
+    if (!createTopic(adminClient, sampleStoreTopic)) {
+      // Update topic config and partition count to ensure desired properties.
+      maybeUpdateTopicConfig(adminClient, sampleStoreTopic);
+      maybeIncreasePartitionCount(adminClient, sampleStoreTopic);
+    }
+  }
+
+  @Override
+  public void storeSamples(MetricSampler.Samples samples) {
+    final AtomicInteger metricSampleCount = new AtomicInteger(0);
+    if (_partitionMetricSampleStoreTopic != null) {
+      for (PartitionMetricSample sample : samples.partitionMetricSamples()) {
+        _producer.send(new ProducerRecord<>(_partitionMetricSampleStoreTopic, null, sample.sampleTime(), null, sample.toBytes()),
+            new Callback() {
+              @Override
+              public void onCompletion(RecordMetadata recordMetadata, Exception e) {
+                if (e == null) {
+                  metricSampleCount.incrementAndGet();
+                } else {
+                  LOG.error("Failed to produce partition metric sample for {} of timestamp {} due to exception",
+                      sample.entity().tp(), sample.sampleTime(), e);
+                }
+              }
+            });
+      }
+    }
+    final AtomicInteger brokerMetricSampleCount = new AtomicInteger(0);
+    if (_brokerMetricSampleStoreTopic != null) {
+      for (BrokerMetricSample sample : samples.brokerMetricSamples()) {
+        _producer.send(new ProducerRecord<>(_brokerMetricSampleStoreTopic, sample.toBytes()), new Callback() {
+          @Override
+          public void onCompletion(RecordMetadata recordMetadata, Exception e) {
+            if (e == null) {
+              brokerMetricSampleCount.incrementAndGet();
+            } else {
+              LOG.error("Failed to produce model training sample due to exception", e);
+            }
+          }
+        });
+      }
+    }
+    _producer.flush();
+    if (LOG.isDebugEnabled()) {
+      LOG.debug("Stored {} partition metric samples and {} broker metric samples to Kafka",
+          metricSampleCount.get(), brokerMetricSampleCount.get());
+    }
+  }
```

Comment:
What is the intention here? Given that this is a draft PR, maybe the implementation of this class is in progress?

Reply:
Removed the broker metrics specific code.